allensdk/brain_observatory/behavior/behavior_session.py

```diff
@@ -1174,33 +1187,164 @@ def metadata(self) -> Dict[str, Any]:
         return self._get_metadata(behavior_metadata=self._metadata)
 
     @classmethod
-    def _read_data_from_stimulus_file(
-            cls, stimulus_file: BehaviorStimulusFile,
+    def _read_licks(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile]) -> Licks:
+        """
+        Construct the Licks data object for this session
+        """
+
+        stimulus_timestamps = cls._read_behavior_stimulus_timestamps(
+                        sync_file=sync_file,
+                        stimulus_file_lookup=stimulus_file_lookup,
+                        monitor_delay=0.0)
+
+        return Licks.from_stimulus_file(
+            stimulus_file=stimulus_file_lookup.behavior_stimulus_file,
+            stimulus_timestamps=stimulus_timestamps)
+
+    @classmethod
+    def _read_rewards(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
+            sync_file: Optional[SyncFile]) -> Rewards:
+        """
+        Construct the Rewards data object for this session
+        """
+        stimulus_timestamps = cls._read_behavior_stimulus_timestamps(
+                        sync_file=sync_file,
+                        stimulus_file_lookup=stimulus_file_lookup,
+                        monitor_delay=0.0)
+
+        return Rewards.from_stimulus_file(
+            stimulus_file=stimulus_file_lookup.behavior_stimulus_file,
+            stimulus_timestamps=stimulus_timestamps.subtract_monitor_delay())
+
+    @classmethod
+    def _read_stimuli(
+            cls,
+            stimulus_file_lookup: StimulusFileLookup,
             stimulus_timestamps: StimulusTimestamps,
```

Comment:
May be better to construct timestamps within this method to make the code symmetric with the other `read*` methods.

Reply:
good idea. Done